src/test/java/org/sagebionetworks/bridge/exporter/integration/ExportTest.java

```diff
@@ -0,0 +1,813 @@
+                .withRecordIdS3Override(s3FileName).build();
+
+        // assert only the first upload was exported
+        assertExport(exporterRequest, null, uploadId, 0, false, true, false);
+    }
+
+    private void verifyExport(String uploadId, int offset, boolean noUpload) throws SynapseException, InterruptedException, IOException {
+        UploadValidationStatus
+                uploadStatus = user.getClient(ForConsentedUsersApi.class).getUploadStatus(uploadId).execute().body();
+        if (!noUpload) {
+            assertEquals(SynapseExporterStatus.SUCCEEDED, uploadStatus.getRecord().getSynapseExporterStatus());
+        } else {
+            assertNull(uploadStatus.getRecord().getSynapseExporterStatus());
+        }
+
+        // then, verify it creates correct synapse table in synapse and correct fields in tables
+        Item appVersionItem = ddbSynapseMetaTables.getItem("tableName", studyId + "-appVersion");
+        Item statusItem = ddbSynapseMetaTables.getItem("tableName", studyId + "-status");
+        Item schemaKeyItem = ddbSynapseTables.getItem("schemaKey", studyId + "-legacy-survey-v1");
+
+        String appVersionTableId = appVersionItem.getString("tableId");
+        String statusTableId = statusItem.getString("tableId");
+        String schemaKeyTableId = schemaKeyItem.getString("tableId");
+
+        if (noUpload) {
+            assertTrue(StringUtils.isBlank(appVersionTableId));
+            assertTrue(StringUtils.isBlank(statusTableId));
+            assertTrue(StringUtils.isBlank(schemaKeyTableId));
+
+        } else {
+            assertFalse(StringUtils.isBlank(appVersionTableId));
+            assertFalse(StringUtils.isBlank(statusTableId));
+            assertFalse(StringUtils.isBlank(schemaKeyTableId));
+
+            Entity appVersionTable = synapseClient.getEntityById(appVersionTableId);
+            assertEquals(TableEntity.class.getName(), appVersionTable.getEntityType());
+
+            assertEquals(project.getId(), appVersionTable.getParentId()); // parent id of a table is project id
+
+            // verify fields
+            Item uploadRecord = ddbRecordTable.getItem("id", uploadStatus.getRecord().getId());
+
+            // app version
+            String jobIdTokenAppVersion = synapseClient.queryTableEntityBundleAsyncStart("select * from " + appVersionTableId, 0L, 100L, true, 0xF, appVersionTableId);
+
+            TimeUnit.SECONDS.sleep(3);
+
+            // study status
+            String jobIdTokenStudyStatus = synapseClient.queryTableEntityBundleAsyncStart("select * from " + statusTableId, 0L, 100L, true, 0xF, statusTableId);
+
+            TimeUnit.SECONDS.sleep(3);
+
+            // survey table
+            String jobIdTokenSurvey = synapseClient.queryTableEntityBundleAsyncStart("select * from " + schemaKeyTableId, 0L, 100L, true, 0xF, schemaKeyTableId);
+
+            // wait for query synapse table
+            TimeUnit.SECONDS.sleep(3);
```

Comment:
3 secs is the minimum wait time i can get. Also, even if we kick off async requests, synapse will not respond any shorter than 9 secs for all of those tasks to complete. I tried to send those request all at once but i can't get all responses within 3 or 6 secs.